src/components/form/SelectPositions/index.tsx

```diff
@@ -0,0 +1,201 @@
+import { TitleValue } from 'components/text/TitleValue'
+import { useBatchBalanceContext } from 'contexts/BatchBalanceContext'
+import { useMultiPositionsContext } from 'contexts/MultiPositionsContext'
+import { useWeb3ConnectedOrInfura } from 'contexts/Web3Context'
+import { BigNumber } from 'ethers/utils'
+import { Position } from 'hooks'
+import React from 'react'
+import { GetMultiPositions_positions } from 'types/generatedGQL'
+
+interface Props {
+  title: string
+  singlePosition?: boolean
+  showOnlyPositionsWithBalance?: boolean
+  callbackToBeExecutedOnRemoveAction?: () => void
+}
+
+const canCanculatePositionToDisplay = (
+  positionsLoading: boolean,
+  balancesLoading: boolean,
+  positions: GetMultiPositions_positions[],
+  balances: BigNumber[],
+  positionIds: string[]
+) => {
+  return (
+    !positionsLoading &&
+    !balancesLoading &&
+    positions.length &&
+    balances.length &&
+    balances.length === positionIds.length &&
+    positions.length === positionIds.length &&
+    JSON.stringify(positions.map(({ id }) => id).sort()) === JSON.stringify([...positionIds].sort())
+  )
+}
+
+export const SelectPositions = ({
+  callbackToBeExecutedOnRemoveAction,
+  showOnlyPositionsWithBalance,
+  singlePosition,
+  title,
+}: Props) => {
+  const { networkConfig } = useWeb3ConnectedOrInfura()
+  const [isModalOpen, setIsModalOpen] = React.useState(false)
+
+  const {
+    errors: positionsErrors,
+    loading: positionsLoading,
+    positionIds,
+    positions,
+    updatePositionIds,
+  } = useMultiPositionsContext()
+
+  const {
+    balances,
+    errors: balancesErrors,
+    loading: balancesLoading,
+    updateBalaces,
+  } = useBatchBalanceContext()
+
+  const [positionsToDisplay, setPositionsToDisplay] = React.useState<Array<string>>([])
+
+  const closeModal = React.useCallback(() => setIsModalOpen(false), [])
+  const openModal = React.useCallback(() => setIsModalOpen(true), [])
+
+  const onModalConfirm = React.useCallback(
+    (positions: Array<Position>) => {
+      const ids = positions.map((position) => position.id)
+      updatePositionIds(ids)
+      updateBalaces(ids)
+      closeModal()
+    },
+    [closeModal, updateBalaces, updatePositionIds]
+  )
+
+  const onRemovePosition = React.useCallback(
+    (positionId: string) => {
+      if (
+        callbackToBeExecutedOnRemoveAction &&
+        typeof callbackToBeExecutedOnRemoveAction === 'function'
+      ) {
+        callbackToBeExecutedOnRemoveAction()
+      }
+      const ids = positionIds.filter((id) => id !== positionId)
+      updatePositionIds(ids)
+      updateBalaces(ids)
+    },
+    [positionIds, updateBalaces, updatePositionIds, callbackToBeExecutedOnRemoveAction]
+  )
+
+  React.useEffect(() => {
+    if (positionIds.length > 0) {
+      if (
+        canCanculatePositionToDisplay(
+          positionsLoading,
+          balancesLoading,
+          positions,
+          balances,
+          positionIds
+        )
```

Comment:
Unless there is something that needs to be done if the first statement is true, regardless of the second, I'd go with both in the same line as it is clearer that both statements need to be true for something to happen

Reply:
This is ok, `positionsIds.length = 0` means that no positions have been selected and `canCanculatePositionToDisplay` checks that everything is in sync to update displayed list